src/osp/tasks/top_session.h

```diff
@@ -55,6 +61,17 @@
 #define OSP_AUX_DGDI_B(x) x
 #define OSP_AUX_DGDI_A(...) OSP_AUX_DGDI_B(OSP_AUX_DGDI_C(__VA_ARGS__))
 
+/**
+ * @brief Retrieves a section of data within topData that was already given to a session
+ * and maps an ID variable to each element that is retrieved.
+ *
+ * Because the data was already given, the indices are stored within the session and topData does
+ * not need to be accessed.
+ *
+ * @param arglist A variadic argument broken into two sections:
```

Comment:
instead of documenting implementation details like the macro taking a variadic argument list, just document what the macro takes.

sounds like the macro simply always requires a count parameter, so probably this OSP_DECLARE_CREATE_DATA_IDS() macros should be given the count parameter as an explicit parameter instead of relying on other macros that it call to break it out of the variadic argument.